src/preact/component/3p-frame.js

```diff
@@ -105,7 +104,7 @@ function ProxyIframeEmbedWithRef(
     };
     setNameAndSrc({
       name: JSON.stringify({
-        'host': parseUrlDeprecated(src).hostname,
+        'host': new URL(src, win.location).hostname,
```

Comment:
Techincally, but does TS not yell at you? The second arg needs to be a string, but `location` gets coerced to a string so it works in regular JS.

Reply:
`preact/component/*.js` isnt' being typechecked yet. that will happen once this lands: https://github.com/ampproject/amphtml/pull/37463

Will preemptively call `.toString()` here.

Reply:
I think you're looking for `.href`… 😝